controllers/cluster_controller.go

```diff
@@ -389,6 +389,10 @@ func (r *ClusterReconciler) preClusterProviderReconcile(ctx context.Context, log
 		return controller.Result{}, err
 	}
 
+	if err := validateExtendedK8sVersionSupport(ctx, r.client, cluster); err != nil {
```

Comment:
Are we planning to update the cluster status in anyway, or this will just be reflected as an error in the controller logs?

Reply:
For now this will just be reflected as an error in the controller logs. I can add it in the cluster status field too. Would you like to make that change in this PR?